### PR TITLE
Add exception to signedOutFallback so signed in users can see the pages

### LINKED ISSUE #4861: Community owners can't preview branded login
You can't preview /:communitySlug/login because our login handler automatically redirects you if you're signed in.

### PR DESCRIPTION
**Pull request by rafaelalmeidatk**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

**Related issues (delete if you don't know of any)**
Closes #4861 

I guess this is a simple enough solution, if there is a `?preview` on the end of `/:communitySlug/login` the user will be able to see the page without being redirected.

### CODE REVIEW

File: src/helpers/signed-out-fallback.js [hunk 1]
```diff
@@ -5,11 +5,27 @@ import { withCurrentUser } from 'src/components/withCurrentUser';
...
+  let isException = false;
+
+  if (searchParamException) {
+    const searchParamsString = (props.location && props.location.search) || '';
+    const searchParams = new Map(
+      searchParamsString
+        .substring(1)
+        .split('&')
+        .map(kv => kv.split('='))
+    );
```
Comment (mxstbr) [inline src/helpers/signed-out-fallback.js]: No need for using a Map here, let's use the `query-string` module we already use across our codebase:

```JS
import queryString from 'query-string';

const searchParams = queryString.parse(searchParamsString);
const isException = searchParams[searchParamsException] && searchParams[searchParamsException] !== false;
```

**Review by mxstbr — COMMENTED** [review]
This makes sense to me overall, I like the simplicity of the approach!

I have one nitpick inline about using the query-string module, but other than that this is great, thanks for looking into this!

File: src/helpers/signed-out-fallback.js (hunk 1 above)
Comment (dev-drprasad) [reply src/helpers/signed-out-fallback.js]: @mxstbr @rafaelalmeidatk i want to add suggestion. The login page still will be functional. we can disable by simply covering whole login page with `overlay` with transparent background. But i am not sure how that fit in this solution

Comment (mxstbr) [reply src/helpers/signed-out-fallback.js]: Honestly for this first version it is probably fine if the login page keeps working 👍

Comment (rafaelalmeidatk) [reply src/helpers/signed-out-fallback.js]: Done, but for some weird reason, I can't log in anymore in my local project so I couldn't test it.

**Pull request closed without merging**